### ISSUE TITLE
Species page links with trailing dot throw error

### ISSUE THREAD

**Issue by nickdos** [AtlasOfLivingAustralia/bie-plugin#147]
@nickdos commented on [Thu Apr 04 2019](https://github.com/AtlasOfLivingAustralia/bie-index/issues/238)

Reported by vjrj on Slack:

> For instance: https://especies.gbif.es/search?q=ginkgo (see first result) generate links like: [https://especies.gbif.es/species/Ginkgo%20L.](https://especies.gbif.es/species/Ginkgo%20L.)

which shows error:

```
Error 500: Internal Server Error
URI
/species/Ginkgo%20L.
Class
java.lang.StringIndexOutOfBoundsException
Message
String index out of range: 1
-----
 Line | Method
->> 1963 | substring                     in java.lang.String
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    256 | getPropertyNameRepresentation in grails.util.GrailsNameUtils
|    215 | getPropertyName . . . . . . . in     ''
|     33 | toUrlElement                  in grails.web.CamelCaseUrlConverter
|    201 | getActionName . . . . . . . . in org.grails.web.mapping.DefaultUrlMappingInfo
|     98 | populateParamsForMapping      in org.grails.web.mapping.AbstractUrlMappingInfo
```

Which suggests this is a Grails issue, related to looking for a file extension (maybe).

**Comment by vjrj** [comment]
Compare:
[https://especies.gbif.es/species/Ginkgo%20biloba%20L.](https://especies.gbif.es/species/Ginkgo%20biloba%20L.)
with the ws:
[https://especies-ws.gbif.es/species/Ginkgo%20biloba%20L.](https://especies-ws.gbif.es/species/Ginkgo%20biloba%20L.)

**Comment by charvolant** [comment]
Being caused by the trailing '.' on the name. For reasons best known to itself, grails is trying to interpret this as a trailing property name of ''

**Comment by charvolant** [comment]
Note the above merge is for something else and me copy/pasting the wrong thing.

**Comment by vjrj** [comment]
Maybe it can be fixed with something like `grails.mime.file.extensions = false` as suggested [here](https://stackoverflow.com/questions/615425/grails-urlmappings-with-parameter-values-containing).

**Comment by vjrj** [comment]
I can confirm that this bug is fixed in `grails 3.2.12`.

I've just tested in `ala-bie` with:
```
diff --git a/gradle.properties b/gradle.properties
index d3a970c..dd596d7 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,5 +1,5 @@
 #Thu Dec 14 10:18:33 AEDT 2017
-grailsVersion=3.2.11
+grailsVersion=3.2.12
 grailsWrapperVersion=1.0.0
 gormVersion=6.0.12.RELEASE
 gradleWrapperVersion=3.4.1
```

More info:
https://github.com/grails/grails-core/issues/10767
